### PR TITLE
Add RakuAST::Type::Subset

### PR DESCRIPTION
**Pull request by ab5tract**
Note: This is currently totally broken. I'm creating the PR so it is easy to discuss the code while learning how to actually accomplish the PR's objective: Add `RakuAST::Type::Subset`. To that extent, it contains debug logging and many, many issues.

### CODE REVIEW

File: src/Perl6/Metamodel/SubsetHOW.nqp [hunk 1]
```diff
@@ -71,6 +71,10 @@ class Perl6::Metamodel::SubsetHOW
         }
     }
 
+    method set_where($obj, $refinement) {
```
Comment (MasterDuke17) [inline src/Perl6/Metamodel/SubsetHOW.nqp:74]: Is ‘$obj’ intentionally ignored?

File: src/Raku/ast/package.rakumod [hunk 2]
```diff
@@ -125,7 +125,7 @@ class RakuAST::Package
             # We always install it as a lexical symbol.
             $resolver.current-scope.merge-generated-lexical-declaration:
                 RakuAST::Declaration::LexicalPackage.new:
-                    :lexical-name($name.canonicalize),
+                    :lexical-name($final),
```
Comment (MasterDuke17) [inline src/Raku/ast/package.rakumod:128]: ‘$final’ only has a value if ‘$name.is-identifier’, is that correct?

Comment (ab5tract) [reply src/Raku/ast/package.rakumod:130]: IIUC, no. `$final` is also set in the else block below: `$final := nqp::pop(@parts).name;`.

**Review by niner — APPROVED** [review]
If you rename IMPL-INSTALL-INTO-PACKAGE this is good to go!

**Review by niner — APPROVED** [review]

**Pull request merged**